vowpalwabbit/global_data.h

```diff
@@ -1,531 +1,535 @@
-
-  // Prediction output
-  std::vector<std::unique_ptr<VW::io::writer>> final_prediction_sink;  // set to send global predictions to.
-  std::unique_ptr<VW::io::writer> raw_prediction;                  // file descriptors for text output.
-
-  VW_DEPRECATED("print has been deprecated, use print_by_ref")
-  void (*print)(VW::io::writer*, float, float, v_array<char>);
-  void (*print_by_ref)(VW::io::writer*, float, float, const v_array<char>&);
-  VW_DEPRECATED("print_text has been deprecated, use print_text_by_ref")
-  void (*print_text)(VW::io::writer*, std::string, v_array<char>);
-  void (*print_text_by_ref)(VW::io::writer*, const std::string&, const v_array<char>&);
-  loss_function* loss;
-
-  VW_DEPRECATED("This is unused and will be removed")
-  char* program_name;
-
-  bool stdin_off;
-
-  bool no_daemon = false;  // If a model was saved in daemon or active learning mode, force it to accept local input when loaded instead.
-
-  // runtime accounting variables.
-  float initial_t;
-  float eta;  // learning rate control.
-  float eta_decay_rate;
-  time_t init_time;
-
-  std::string final_regressor_name;
-
-  parameters weights;
-
-  size_t max_examples;  // for TLC
-
-  bool hash_inv;
-  bool print_invert;
-
-  // Set by --progress <arg>
-  bool progress_add;   // additive (rather than multiplicative) progress dumps
-  float progress_arg;  // next update progress dump multiplier
-
-  std::map<uint64_t, std::string> index_name_map;
-
-  label_type_t label_type;
-
-  vw();
-  ~vw();
-  std::shared_ptr<rand_state> get_random_state() { return _random_state_sp; }
-
-  vw(const vw&) = delete;
-  vw& operator=(const vw&) = delete;
-
-  // vw object cannot be moved as many objects hold a pointer to it.
-  // That pointer would be invalidated if it were to be moved.
-  vw(const vw&&) = delete;
-  vw& operator=(const vw&&) = delete;
-};
-
-VW_DEPRECATED("Use print_result_by_ref instead")
-void print_result(VW::io::writer* f, float res, float weight, v_array<char> tag);
-void print_result_by_ref(VW::io::writer* f, float res, float weight, const v_array<char>& tag);
-
-VW_DEPRECATED("Use binary_print_result_by_ref instead")
-void binary_print_result(VW::io::writer* f, float res, float weight, v_array<char> tag);
-void binary_print_result_by_ref(VW::io::writer* f, float res, float weight, const v_array<char>& tag);
-
-void noop_mm(shared_data*, float label);
-void get_prediction(VW::io::reader* f, float& res, float& weight);
-void compile_gram(
-    std::vector<std::string> grams, std::array<uint32_t, NUM_NAMESPACES>& dest, char* descriptor, bool quiet);
-void compile_limits(std::vector<std::string> limits, std::array<uint32_t, NUM_NAMESPACES>& dest, bool quiet);
-
-VW_DEPRECATED("Use print_tag_by_ref instead")
-int print_tag(std::stringstream& ss, v_array<char> tag);
-int print_tag_by_ref(std::stringstream& ss, const v_array<char>& tag);
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
```

Comment:
Looks like a whitespace change leaves the whole file as modified. Can you please fix this?

Reply:
Where is the extra white space?  I can't find it.

Reply:
Sorry I meant line endings

Reply:
it's probably a windows style vs unix style line endings, so whatever is on this branch it probably is the opposite? depending on your git config it might muck with them too so if you did initially this change in another computer it might be a bit harder to decipher.

Reply:
Got it..  Will check out the line endings.  Thanks.

Reply:
Done.